### ISSUE TITLE
Methods in `org.jboss.eap.qe.ts.common.docker.Docker` tool mask exceptions

### ISSUE THREAD

**Issue by honza-kasik** [jboss-eap-qe/eap-microprofile-test-suite#194]
Almost every method in `org.jboss.eap.qe.ts.common.docker.Docker` has `throws Exception` in its signature which causes masking of exception and leaves no precise information what kind of exception method is throwing.

**Comment by mnovak1** [comment]
afaict when any specific exception is thrown then it's not generalized as `Exception`, user will always see the specific exception. Only issue is with method signatures where the motivation to throw general `Exception` was to keep it simple avoid API breakage in the future. Or do I understand it wrong?

**Comment by honza-kasik** [comment]
It is a bad for anyone who wants to expand current methods. IDE will not remind him that an exception is thrown by newly added code which he might want to catch. It is also bad for anyone who want to use these methods in his code because using them effectively masks any other exception in code and then it is similar to first case. Using `throws Exception` makes risk of introducing an error to the code much higher.

We can use checked exceptions instead. Such as "ContainerStartException" which will be the only exception thrown by the `start` method to avoid API break in future and not masking other exceptions.

**Comment by mnovak1** [comment]
> We can use checked exceptions instead. Such as "ContainerStartException" which will be the only exception thrown by the start method to avoid API break in future and not masking other exceptions.

Right, `Exception`  is too generic without a way to catch anything specific. I'm ok with this change.
